pythonx/jedi_vim_debug.py

```diff
@@ -1,57 +1,78 @@
 """Used in jedi-vim's jedi#debug_info()"""
+import sys
+
 import vim
+from jedi_vim import PythonToVimStr
 
 
 def echo(msg):
-    vim.command('echo {0}'.format(msg))
+    vim.command('echo %r' % PythonToVimStr(msg))
+
+
+def echo_error(msg):
+    vim.command('echohl ErrorMsg')
+    echo(msg)
+    vim.command('echohl None')
+
+
+def format_exc_info(exc_info=None, tb_indent=2):
+    import traceback
+
+    if exc_info is None:
+        exc_info = sys.exc_info()
+
+    exc_msg = traceback.format_exception_only(exc_info[0], exc_info[1])
+    lines = ''.join(exc_msg).rstrip('\n').split('\n')
+
+    lines.append('Traceback (most recent call last):')
+    tb = traceback.format_tb(exc_info[2])
+    lines.extend(''.join(tb).rstrip('\n').split('\n'))
+
+    indent = ' ' * tb_indent
+    return '{0}'.format(('\n' + indent).join(lines))
 
 
 def display_debug_info():
-    echo("printf(' - global sys.version: `%s`', {0!r})".format(
+    echo(' - global sys.version: `{0}`'.format(
         ', '.join([x.strip()
-                   for x in __import__('sys').version.split('\n')])))
-    echo("printf(' - global site module: `%s`', {0!r})".format(
-        __import__('site').__file__))
+                   for x in sys.version.split('\n')])))
+    echo(' - global site module: `{0}`'.format(__import__('site').__file__))
 
     try:
         import jedi_vim
-    except Exception as e:
-        echo("printf('ERROR: jedi_vim is not available: %s: %s', "
-             "{0!r}, {1!r})".format(e.__class__.__name__, str(e)))
+    except Exception:
+        echo_error('ERROR: jedi_vim is not available: {0}'.format(
+            format_exc_info(sys.exc_info())))
```

Comment:
We can probably just remove sys.exc_info here, right?

Reply:
Yeah, I've added the default handling afterwards apparently.

Reply:
Fixed.